src/create-text-tracks-if-necessary.js

```diff
@@ -16,14 +16,36 @@ const createTextTracksIfNecessary = function(sourceBuffer, mediaSource, segment)
 
   // create an in-band caption track if one is present in the segment
   if (segment.captions &&
-      segment.captions.length &&
-      !sourceBuffer.inbandTextTrack_) {
-    removeExistingTrack(player, 'captions', 'cc1');
-    sourceBuffer.inbandTextTrack_ = player.addRemoteTextTrack({
-      kind: 'captions',
-      label: 'cc1'
-    }, false).track;
-    player.tech_.trigger({type: 'usage', name: 'hls-608'});
+      segment.captions.length) {
+    if (!sourceBuffer.inbandTextTracks_) {
+      sourceBuffer.inbandTextTracks_ = {};
+    }
+
+    segment.captions.forEach(function(caption) {
```

Comment:
This will have to loop through every caption of every segment loaded even after the text tracks have been created and added to the player. Though probably not a big deal, seems like a lot of wasted computation. Is there anyway we can attach to the transmuxers output the information of what streams the captions array uses so you can check whether theyve already been made without having to loop through the entire captions array again?